### DISCUSSION TITLE
Data export/import

### DISCUSSION THREAD

**Discussion by Hatchibombotar** [Mojang/minecraft-editor#153] [Feedback]
Use cases:
- Exporting colour pallet data, to be used in different worlds or to give to other people
- Exporting structures

This roughly possible through bds and the server-net module, however this is uncomfortable. Even just exporting json data would be very useful.

**Comment by Wabbabrick** [comment]
Adding a Local Asset Library / Cross-Project Structure Support

I strongly support this request. One major workflow bottleneck right now is moving assets between different Editor projects.

I am really looking forward to a native option to efficiently export and import structures between projects without the manual "Structure Block" export/import dance. Ideally, the Editor could implement a wrapper around .mcstructure files that allows direct read/write access to a local folder on the disk.

This would effectively allow us to build a persistent "Asset Library" (trees, houses, custom props) that lives on our hard drive and can be instantly dragged and dropped into any new project we open.

**Reply by m-brisch** [reply]
Great idea! We're thinking about ways to make it easier to move assets between projects but it'll be a while before we'd be able to release it.

One work-around in the meantime is that if you have structures you author in Editor they'll appear within a `editor/structuredb` folder inside the world folder. If you copy those files between projects they'll be available the next time you load the world. Same applies for other persisted state like jigsaws/layouts/etc.
